Add a checkbox for the textarea field to use wpautop()

kienstra opened the pull request:
This will enable displaying the Textarea control on the front-end like how it looks in the block editor.

Maybe I didn't understand #252 right, or I couldn't reproduce it.

But it looked like before, newlines weren't actually stripped. But multiple lines of text still appeared on one line:

<img width="959" alt="newlines-here" src="https://user-images.githubusercontent.com/4063887/57362628-1cfb9a80-7134-11e9-8f0e-ae0db12e80af.png">

This addresses that by adding a new checkbox to apply `wpautop()`:

<img width="1165" alt="new-checkbox" src="https://user-images.githubusercontent.com/4063887/57362797-65b35380-7134-11e9-9537-5740710070ef.png">

Still, we may want to just let the Rich Text control handle this (#20).

# Before
![before-not-different-lines](https://user-images.githubusercontent.com/4063887/57362393-95ae2700-7133-11e9-832d-8dc2533c649b.gif)

# After 
![after-different-lines](https://user-images.githubusercontent.com/4063887/57362434-ab235100-7133-11e9-9895-6c491c907e2f.gif)

Closes #252

kienstra committed:
Add a checkbox for the textarea field to use wpautop()

This will enable displaying on the front-end
like how this looks in the block editor.

kienstra commented:
**Request For Review**

Hi @lukecarbis,
Could you please review this? Though maybe we could just let the Rich Text control handle this.

Thanks, Luke!

lukecarbis committed:
Add multiple new line format options, and move tax and post specific methods to their own class

lukecarbis committed:
Update unit tests

lukecarbis committed:
Add Control_Setting() to textarea unit test

lukecarbis commented:
This is great, Ryan. I change the checkbox to a select dropdown, so you could choose the formatting option you'd prefer (auto paragraph, auto line-break, or no formatting).

As part of this, I also moved the post and taxonomy specific methods out of the `Control_Abstract`, and into their own controls (since these methods don't need to be shared across multiple controls).

Please review?

lukecarbis commented:
![Screen Recording 2019-05-15 at 11 13 16 am 2019-05-15 11_46_05](https://user-images.githubusercontent.com/1097667/57743087-0f7f7b00-7707-11e9-980a-46768b69787c.gif)

kienstra committed:
Fix a mistake I had made earlier with a unit test

Prepend test_ to function name

This ensures that it is run in unit tests,
otherwise I don't think it will run.
Also, edit the test so that it passes.

kienstra reviewed (commented):
**Approved**

Hi @lukecarbis,
This looks good. The formatting on the front-end looks good in all 3 cases:

# Editor
<img width="1426" alt="formatting-options" src="https://user-images.githubusercontent.com/4063887/57976722-9012d580-79ac-11e9-8801-929c880194eb.png">


# Front-end (ServerSideRender)
<img width="1425" alt="formatting-display-here" src="https://user-images.githubusercontent.com/4063887/57976724-95702000-79ac-11e9-9fde-50130154b115.png">

kienstra commented:
Hi @lukecarbis,
It looks like I can't officially approve this, as I opened it. But it looks good.

Should I merge it if/when Travis passes? The [last commit](https://github.com/getblocklab/block-lab/pull/302/commits/1e728e19cfffee85c157438120eb2a7fffdcda14) is just fixing an error I noticed that I had made earlier. 

It's not really related to this PR, it's just that this PR moved the test to another file.

kienstra commented:
If so, could you please approve this again? It looks like I dismissed your review with that commit.

lukecarbis commented:
All good. I think it auto dismisses if you add a new commit. Ready for you to merge!

kienstra commented:
Thanks!
